src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseSpanClearInsteadOfFill.cs

```diff
@@ -0,0 +1,155 @@
+                operationContext =>
+                {
+                    var invocation = (IInvocationOperation)operationContext.Operation;
+
+                    if (!SymbolEqualityComparer.Default.Equals((invocation.Instance.Type as INamedTypeSymbol)?.ConstructedFrom, spanType))
+                    {
+                        return;
+                    }
+
+                    if (invocation.TargetMethod.Name != FillMethod)
+                    {
+                        return;
+                    }
+
+                    if (invocation.Arguments.Length != 1)
+                    {
+                        return;
+                    }
+
+                    if (IsDefaultValue(invocation.Arguments[0]))
+                    {
+                        operationContext.ReportDiagnostic(Diagnostic.Create(s_Rule, invocation.Syntax.GetLocation()));
+                    }
+                },
+                OperationKind.Invocation);
+        }
+
+        private static bool IsDefaultValue(IArgumentOperation argumentOperation)
+        {
+            var value = argumentOperation.Value;
+            var type = argumentOperation.Parameter.Type;
+            var constantOpt = value.ConstantValue;
+
+            if (constantOpt.HasValue)
+            {
+                if (constantOpt.Value == null)
+                {
+                    // null must be default value for any valid type
+                    return true;
+                }
+
+                if (argumentOperation.Type.IsNullableValueType())
+                {
+                    // 0 isn't default value for T?
+                    return false;
+                }
+
+                // enum/nint are treated as integers
+                // This is missing default value of DateTime literal for VB,
+                // but since VB doesn't properly support Span, just don't consider it
+                switch (constantOpt.Value)
+                {
+#pragma warning disable IDE0004 // Remove Unnecessary Cast - false positive
+                    case (byte)0 or (short)0 or (int)0 or (long)0 or
+                        (sbyte)0 or (ushort)0 or (uint)0 or (ulong)0:
+                        return true;
+#pragma warning restore IDE0004 // Remove Unnecessary Cast
+
+                    // -0 is not all bits zero. Handle them by bits
+                    case float f:
+                        // SingleToInt32Bits not available in netstandard2.0
+                        return BitConverter.DoubleToInt64Bits(f) == 0;
+
+                    case double d:
+                        return BitConverter.DoubleToInt64Bits(d) == 0;
+
+                    case decimal d:
+                        return decimal.GetBits(d).All(b => b == 0);
+
+                    default:
+                        return false;
+                }
+            }
+
+            if (value is IDefaultValueOperation)
+            {
+                return SymbolEqualityComparer.Default.Equals(value.Type, type);
```

Comment:
Probably will cause false negative for the following code?

```csharp
using System;
public class Base {}
public class Derived : Base {}

public class C
{
    public void M(Span<Base> span)
    {
        span.Fill(default(Derived));
    }
}
```

Reply:
This will get `null` in constant value and handled earlier.

Reply:
Please do add a test for this case, if you haven't already done so. Thanks!